### PR TITLE
ci: try cibuildwheel pyodide

### PR DESCRIPTION
**Pull request by henryiii**
Try https://github.com/pypa/cibuildwheel/pull/1456.

### CODE REVIEW

File: .github/workflows/emscripten.yaml [hunk 1]
```diff
@@ -46,7 +46,7 @@ jobs:
           actions-cache-folder: emsdk-cache
 
       - name: Build
-        run: CFLAGS=-fexceptions LDFLAGS=-fexceptions pyodide build  --exports whole_archive
+        run: cflags=-fexceptions ldflags=-fexceptions pyodide build  --exports whole_archive
```
Comment (henryiii) [inline .github/workflows/emscripten.yaml:49]:
```suggestion
        run: CFLAGS=-fexceptions LDFLAGS=-fexceptions pyodide build  --exports whole_archive
```

**Comment by hoodmane** [comment]
I am successfully reproducing this locally, thanks @henryiii!

**Comment by hoodmane** [comment]
Is there an easy way to make cibuildwheel leave its temp directories around after it is done?

**Comment by henryiii** [comment]
You can see the error without cibuildwheel, see #938.

File: pyproject.toml [hunk 2]
```diff
@@ -156,7 +156,7 @@ ignore_missing_imports = true
 
 [tool.cibuildwheel]
 test-extras = "test"
-test-command = "pytest --benchmark-disable {project}/tests"
+test-command = "python -m pytest --benchmark-disable {project}/tests"
```
Comment (henryiii) [inline pyproject.toml:159]:
```suggestion
test-command = "pytest --benchmark-disable {project}/tests"
```

File: .github/workflows/emscripten.yaml [hunk 3]
```diff
@@ -65,3 +65,26 @@ jobs:
...
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+          fetch-depth: 0
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+
+      - uses: hoodmane/cibuildwheel@emscripten
```
Comment (henryiii) [inline .github/workflows/emscripten.yaml:82]:
```suggestion
      - uses: pypa/cibuildwheel@main
```

File: .github/workflows/emscripten.yaml [hunk 4]
```diff
@@ -26,10 +26,10 @@ jobs:
 
       - uses: actions/setup-python@v5
         with:
-          python-version: "3.11"
+          python-version: "3.12"
 
       - name: Install pyodide-build
-        run: pip install pyodide-build==0.25.1
+        run: pip install pyodide-build==0.26.0a6
```
Comment (henryiii) [inline .github/workflows/emscripten.yaml:32]:
```suggestion
        run: pip install pyodide-build==0.26.0
```

File: .github/workflows/emscripten.yaml [hunk 5]
```diff
@@ -58,3 +58,22 @@ jobs:
...
+  build-pyodide:
+    name: Pyodide cibuildwheel
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+          fetch-depth: 0
+
+      - uses: pypa/cibuildwheel@v2.19
```
Comment (henryiii) [inline .github/workflows/emscripten.yaml:71]:
```suggestion
      - uses: pypa/cibuildwheel@henryiii-patch-1
```

**Pull request merged**